pkg/handlers/v1/notification.go

```diff
@@ -0,0 +1,71 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-scan-notifier/pkg/domain"
+	"github.com/asecurityteam/nexpose-scan-notifier/pkg/logs"
+)
+
+// Output contains a list of completed Nexpose scans.
+type Output struct {
+	Response []scanNotification `json:"response"`
+}
+
+// scanNotification represents
+type scanNotification struct {
+	ScanID string `json:"scanID"`
+	SiteID string `json:"siteID"`
+}
+
+// NotificationHandler takes a duration and returns a list of completed scans.
+type NotificationHandler struct {
+	ScanFetcher      domain.ScanFetcher
+	TimestampFetcher domain.TimestampFetcher
+	TimestampStorer  domain.TimestampStorer
+	Producer         domain.Producer
+	LogFn            domain.LogFn
+}
+
+// Handle accepts takes Input with a duration, queries for completed scans within the duration,
+// produces all completed scans to a queue, and returns the list of completed scans.
+func (h *NotificationHandler) Handle(ctx context.Context) (Output, error) {
+	logger := h.LogFn(ctx)
+
+	lastScanTimestamp, err := h.TimestampFetcher.FetchTimestamp(ctx)
+	switch err.(type) {
+	case nil:
```

Comment:
Question/Opinion: Do you need this line here checking for a nil error? I've seen it this way in the past and I think it's a little more clear:

`lastScanTimestamp, err := h.TimestampFetcher.FetchTimestamp(ctx)
if err != nil {   //don't even check the error types if err is nil
    switch err.(type) {
    case domain.TimestampNotFound:
    default:
        logger.Error(logs.StorageFailure{Reason: err.Error()})
	    return Output{}, err
    }
}
...
`

Reply:
the way that it is written is how it usually appears in the go documentation: https://golang.org/ref/spec#TypeSwitchStmt

It also removes an additional level of indentation, which seems to be the general preference in Go code. From [go best practices](https://talks.golang.org/2013/bestpractices.slide#4):

> Less nesting means less cognitive load on the reader

Reply:
True the docs show a nil case, but it's never empty in those examples. What's strange about this is that the case is empty, not that it's checking for nil. We want the nil case to fall through, which to me seems like it shouldn't be what we check for and instead we should check for the opposite, but we can't check for `!nil` because go doesn't like that syntax, so I feel like using the more traditional way to check for errors then handling the error in the nil-check block is more readable.  Though in the end readability is a preference, so don't let this block the PR.